flow/layers/physical_shape_layer.cc

```diff
@@ -142,14 +138,15 @@ void PhysicalShapeLayer::Paint(PaintContext& context) const {
   // Frame) and once into the Frame's texture (which is then drawn on top of the
   // current canvas).
   if (kRenderPhysicalShapeUsingSystemCompositor) {
-    // If we are being rendered into our own frame using the system compositor,
-    // then it is neccesary to "punch a hole" in the canvas/frame behind us so
-    // that group opacity looks correct.
+#if defined(OS_FUCHSIA)
+    // On Fuchsia, If we are being rendered into our own frame using the system
+    // compositor, then it is neccesary to "punch a hole" in the canvas/frame
+    // behind us so that single-pass group opacity looks correct.
     SkPaint paint;
     paint.setColor(SK_ColorTRANSPARENT);
     paint.setBlendMode(SkBlendMode::kSrc);
-    context.leaf_nodes_canvas->drawRect(paint_bounds(), paint);
-
+    context.internal_nodes_canvas->drawRect(paint_bounds(), paint);
```

Comment:
Apologize if I forget something that we discussed before: why is this using `internal_nodes_canvas` instead of `leaf_nodes_canvas`?

Reply:
Amir mentioned that ContainerLayer's render into internal_nodes_canvas when I spoke to him about how composition works.  However, that is only used for iOS so this should probably go to leaf_node_canvas.

Reply:
(didn't review the entire PR so I may be missing some context).
Generally draws should always be done on the leaf_nodes_canvas, the internal_nodes_canvas forwards the operation to all canvases for all composited surfaces.